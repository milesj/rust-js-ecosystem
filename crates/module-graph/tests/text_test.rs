mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::generate_graph_for_file;

mod text {
    use super::*;

    #[test]
    fn graphql() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.gql"));
    }

    #[test]
    fn graphql_alt_extension() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "graphql.graphql"));
    }

    #[test]
    fn html() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "index.html"));
    }

    #[test]
    fn less() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.less"));
    }

    #[test]
    fn map() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "source.js.map"));
    }

    #[test]
    fn sass() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.sass"));
    }

    #[test]
    fn scss() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.scss"));
    }

    #[test]
    fn stylus() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "styles.styl"));
    }

    #[test]
    fn svg() {
        let sandbox = create_sandbox("text");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "icon.svg"));
    }
}
