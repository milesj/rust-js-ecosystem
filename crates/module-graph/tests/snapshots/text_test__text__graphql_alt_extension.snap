---
source: crates/module-graph/tests/text_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"graphql.graphql\")"
---
{
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [],
        package_name: None,
        path: "/root/graphql.graphql",
        query: None,
        source: TextModule {
            kind: Graphql,
            source: "type Query {\n  hero: Character\n}\n\ntype Character {\n  name: String\n  friends: [Character]\n  homeWorld: Planet\n  species: Species\n}\n\ntype Planet {\n  name: String\n  climate: String\n}\n\ntype Species {\n  name: String\n  lifespan: Int\n  origin: Planet\n}\n",
        },
    },
}
